### DISCUSSION TITLE
Q: 评论区的上传图片功能如何接入自己的图床 || Q: How to connect the image upload function in the comment area to your own image bed

### DISCUSSION THREAD

**Discussion by shulng** [walinejs/waline#2595] [Q&A]
### Describe the bug | 问题描述

我用的图床项目是https://github.com/x-dr/telegraph-Image

### Website URL | 问题网站

https://hexo-image.pages.dev/

### Where your waline deploy? | 服务部署在哪里？

Vercel (Default)

### Where your comment data store? | 数据存储在哪里？

LeanCloud(https://leancloud.app)

---
### Describe the bug | Problem description

The image bed project I use is https://github.com/x-dr/telegraph-Image

### Website URL | Problem website

https://hexo-image.pages.dev/

### Where your waline deploy? | Where is the service deployed?

Vercel (Default)

### Where your comment data store? | Where is the data stored?

LeanCloud(https://leancloud.app)

**Comment by lizheming** [comment]
https://waline.js.org/cookbook/customize/upload-image.html

**Reply by shulng** [reply]
试过貌似不行

---
Tried but it doesn’t seem to work

**Comment by huochairener** [comment]
这个我实现了，我用的是butterfly主题。先搭建个兰空图床，然后修改waline.pug文件，这两个步骤我的博客里都有写，搭建兰空图床：https://www.huochairener-blog.cn/2024/11/06/%E6%90%AD%E5%BB%BA%E5%85%B0%E7%A9%BA%E5%9B%BE%E5%BA%8A%E5%B9%B6%E9%85%8D%E5%90%88PicGo%E5%AE%9E%E7%8E%B0%E6%89%B9%E9%87%8F%E4%B8%8A%E4%BC%A0/
waline更换图床：https://www.huochairener-blog.cn/2024/07/11/%E4%B8%BAHexo%E6%B7%BB%E5%8A%A0%E8%AF%84%E8%AE%BA%E7%B3%BB%E7%BB%9FWaline/
当时写文章的时候用的是去不图床，不过去不其实用的也是兰空图床，自己搭建完了兰空图床后就改改参数值就行，参数名都一样。

---
I implemented this, using the butterfly theme. First build a Lankong picture bed, and then modify the waline.pug file. These two steps are written in my blog. Build a Lankong picture bed: https://www.huochairener-blog.cn/2024/11/06 /%E6%90%AD%E5%BB%BA%E5%85% B0%E7%A9%BA%E5%9B%BE%E5%BA%8A%E5%B9%B6%E9%85%8D%E5%90%88PicGo%E5%AE%9E%E7%8E%B0% E6%89%B9%E9%87%8F%E4%B8%8A%E4%BC%A0/
waline replacement picture bed: https://www.huochairener-blog.cn/2024/07/11/%E4%B8%BAHexo%E6%B7%BB%E5%8A%A0%E8%AF%84%E8% AE%BA%E7%B3%BB%E7%BB%9FWaline/
When I was writing the article, I used the Gobu map bed, but the practical one was also the Lankong map bed. After I built the Lankong map bed, I just changed the parameter values. The parameter names are the same.

**Reply by huochairener** [reply]
我用的Hexo的Butterfly主题，看看有没有帮助

---
I use Hexo’s Butterfly theme, see if it helps
